Gulpfile.ts

```diff
@@ -0,0 +1,1125 @@
+        { target: "lib.es2016.d.ts", sources: ["header.d.ts", "es2016.d.ts"] },
+        { target: "lib.es2017.d.ts", sources: ["header.d.ts", "es2017.d.ts"] },
+
+        // JavaScript + all host library
+        { target: "lib.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(hostsLibrarySources) },
+        { target: "lib.es6.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(es2015LibrarySources, hostsLibrarySources, "dom.iterable.d.ts") }
+].concat(es2015LibrarySourceMap, es2016LibrarySourceMap, es2017LibrarySourceMap);
+
+const libraryTargets = librarySourceMap.map(function (f) {
+    return path.join(builtLocalDirectory, f.target);
+});
+
+for (const i in libraryTargets) {
+    const entry = librarySourceMap[i];
+    const target = libraryTargets[i];
+    const sources = [copyright].concat(entry.sources.map(function (s) {
+        return path.join(libraryDirectory, s);
+    }));
+    gulp.task(target, false, [], function() {
+        if (!needsUpdate(sources, target)) {
+            return gulp.src(target);
+        }
+        return gulp.src(sources).pipe(concat(target, {newLine: ""})).pipe(gulp.dest("."));
+    });
+}
+
+const configureNightlyJs = path.join(scriptsDirectory, "configureNightly.js");
+const configureNightlyTs = path.join(scriptsDirectory, "configureNightly.ts");
+const packageJson = "package.json";
+const programTs = path.join(compilerDirectory, "program.ts");
+
+function needsUpdate(source: string | string[], dest: string | string[]): boolean {
+    if (typeof source === "string" && typeof dest === "string") {
+        if (fs.existsSync(dest)) {
+            const {mtime: outTime} = fs.statSync(dest);
+            const {mtime: inTime} = fs.statSync(source);
+            if (+inTime <= +outTime) {
+                return false;
+            }
+        }
+    }
+    else if (typeof source === "string" && typeof dest !== "string") {
+        const {mtime: inTime} = fs.statSync(source);
+        for (const filepath of dest) {
+            if (fs.existsSync(filepath)) {
+                const {mtime: outTime} = fs.statSync(filepath);
+                if (+inTime > +outTime) {
+                    return true;
+                }
+            }
+            else {
+                return true;
+            }
+        }
+        return false;
+    }
+    else if (typeof source !== "string" && typeof dest === "string") {
+        if (fs.existsSync(dest)) {
+            const {mtime: outTime} = fs.statSync(dest);
+            for (const filepath of source) {
+                if (fs.existsSync(filepath)) {
+                    const {mtime: inTime} = fs.statSync(filepath);
+                    if (+inTime > +outTime) {
+                        return true;
+                    }
+                }
+                else {
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
+    else if (typeof source !== "string" && typeof dest !== "string") {

```

Comment:
I don't see a call site that uses this case, but if there was do you expect it to compare files on a one to one basis?